packages/image/src/extractColors.ts

```diff
@@ -0,0 +1,78 @@
+interface PaletteOptions {
+  colorCount?: number;
+  quality?: number;
+}
+
+/**
+ * Extracts a palette of colors from a given image URL based on provided options.
+ * @param {string} imageUrl - The URL of the image.
+ * @param {PaletteOptions} [options] - The extraction options.
+ */
+export const extractColors = (imageUrl: string, options?: PaletteOptions) => {
+  return loadImageFromUrl(imageUrl)
+    .then(extractImage)
+    .then((data) => {
+      if (!data) {
+        throw new Error(
+          "Failed to extract pixel data from the image. Ensure the image URL is valid and accessible.",
+        );
+      }
+
+      const pixels = filterRelevantPixels(data, options?.quality);
+      const count = options?.colorCount || pixels.length;
+      return pixels.slice(0, count);
+    });
+};
+
+/**
+ * Extracts pixel data from an image.
+ * @param {HTMLImageElement} img - The image element.
+ */
+const extractImage = (img: HTMLImageElement) => {
+  const canvas = document.createElement("canvas");
+  const context = canvas.getContext("2d");
+
+  if (!context) {
+    return null;
+  }
+
+  canvas.width = img.naturalWidth;
+  canvas.height = img.naturalHeight;
+  context.drawImage(img, 0, 0);
+
+  return context.getImageData(0, 0, canvas.width, canvas.height).data;
+};
+
+/**
+ * Loads an image from a given URL.
+ * @param {string} url - The URL of the image to load.
+ */
+const loadImageFromUrl = (url: string) => {
+  return new Promise<HTMLImageElement>((resolve, reject) => {
+    const img = new Image();
+    img.onload = () => resolve(img);
+    img.onerror = reject;
+    img.crossOrigin = "Anonymous";
+    img.src = url;
+  });
+};
```

Comment:
I think it can be replaced by `load` with some modifying itself
or load may be available refer this function. I think we need to reduce repetitive code

Reply:
Of course! The relevant matter has already been raised as an issue #35, so can I proceed directly with the next task? I've come up with a great idea regarding the load function.